docs/proposals/002-api-proposal/proposal.md

```diff
@@ -0,0 +1,313 @@
+        // The name of the model as the users set in the "model" parameter in the requests.
+        // The model name should be unique among the use cases that reference the same backend pool.
+        // This is the parameter that will be used to match the request with. In the future, we may
+        // allow to match on other request parameters. The other approach to support matching on 
+        // on other request parameters is to use a different ModelName f HTTPFilter 
+        ModelName string
+        // Optional
+        // Use cases with an objective have higher priority than use cases without.
+        // IMPORTANT: By specifying an objective, this places the UseCase in a higher priority class than UseCases without a defined priority class. 
+        // In the face of resource-scarcity. Higher priority requests will be preserved, and lower priority class requests will be rejected.
+        Objective *Objective
+        // Optional.
+	// Allow multiple versions of a model for traffic splitting. 
+	// If not specified, the target model name is defaulted to the modelName parameter.
+        TargetModels []common.TargetModel
+}
+
+
+
+// TargetModel represents a deployed model or a LoRA adapter. The
+// TargetModelName is expected to match the name of the LoRA adapter
+// (or base model) as it is registered within the model server. Inference
+// Gateway assumes that the model exists on the model server and is the
+// responsibility of the user to validate a correct match. Should a model fail
+// to exist at request time, the error is processed by the Instance Gateway,
+// and then emitted on the appropriate ModelGroup object.
+type TargetModel struct {
+        // The name of the adapter as expected by the ModelServer.
+        TargetModelName string
+        // Weight is used to determine the percentage of traffic that should be 
+        // sent to this target model when multiple versions of the model are specified.
+        Weight int
+}
+
+// Objective captures the latency SLO of a LLM use case.
+// In MVP, meeting the SLO is on a best effort basis.
+// Future: Extend the API for different behaviors of meeting the SLO.
+// The gateway will perform best-effort load balancing, and work with other components (e.g., autoscaler) to meet the
+// objectives. 
+type Objective struct {
+        // The AverageLatencyPerOutputToken is calculated as the e2e request latency divided by output token 
+        // length. Note that this is different from what is known as TPOT (time per output token) which only 
+        // takes decode time into account.
+        // The P95 is calculated over a fixed time window defined at the operator level.
+        DesiredAveragePerOutputTokenLatencyAtP95OverMultipleRequests
+    *time.Duration
+}
+```
+
+### Yaml Examples
+
+#### BackendPool(s)
+Here we create 2 BEPs that subscribe to services to collect the appropriate pods
+```yaml
+apiVersion: inference.x-k8s.io/v1alpha1
+kind: BackendPool
```

Comment:
Do we plan to state on a CR definition of the BackendPool at the end of this proposal. Or do we plan to open another proposal? ---which seems odd to me.

I'd suggest the following definition based on this yaml:
```golang
// BackendPool represents a set of model servers having a common set useCases.
// BackendPool name is used to uniquely identify the backend pool in the ModelGroup.
type BackendPool struct {
	metav1.ObjectMeta
	metav1.TypeMeta

	Spec BackendPoolSpec
}

type BackendPoolSpec struct {
	// Services references all the available services within this BackendPool.
	Services []v1.ObjectReference `json:"services"`
}
```
Just rough idea.

Reply:
Yeah, I had the BEP definition in here at one point, but removed based on discussion. I meant to add it back, will do so today and update here.

Reply:
Added back

Reply:
You rock!